demoplugin/demo.go

```diff
@@ -30,16 +38,18 @@ func init() {
 				&gain,
 			},
 			ProcessDoubleFunc: func(in, out vst2.DoubleBuffer) {
+				var g = math.Pow(10.0, float64(gain.GetVal())/20.0)
```

Comment:
Why not include this calculation into `GetValue` function? So it'll look like: `math.Pow(10, (-20 + 40 * value) / 20.0)` and here we can just call it? Obsolete zeroes can be truncated in the `GetParamDisplayFunc` format function?

Also, you can use `math.Pow10`.

Reply:
Math.Pow10 have int as parameter and i need float here.

Reply:
I'm convert the parameter value (-20 db to +20 db) back to a float value. Maybe this can be improved more.

Reply:
`GetValueFunc` is  used to convert the default 0 to 1 float Value to something like -20 to 20. 
`GetValueLabelFunc` is used to convert the Value to a string. 
`GetValueFunc` isused here to remove duplicate code. So you don't have to write "-20 + (40 * value)" in each Process function and `GetValueLabelFunc`. Its used to reduce duplicate code here. Maybe i'm overcomplicate it here. But these functions are optional.

Reply:
Thanks for a thorough explanation. I had the following in mind:

1. `Parameter.Value` field holds value in [0;1] interval, host reads it through `getParameterPluginBridge` function;
2. `Parameter.GetValue` checks if `GetValueFunc != nil` and calls it, otherwise `Parameter.Value` is returned; only used by `Plugin.ProcessXXX` function;
3. `Parameter.GetValueFunc` is used to provide mapping of `[0;1]` interval to any required interval (logarithmic/linear/etc);
4. `Parameter.GetValueLabel` is used to get display value, that would be rendered by DAW. It checks if `Parameter.GetValueLabelFunc != nil`  and calls it, otherwise `Parameter.GetValue` is returned.

I think the main problem with demoplugin right now is that we're using linear scale for DB, which isn't right :)

Reply:
Ideally `GetValueFunc` would return logarithmic value that would be used to calculate sample values in `Process` functions and `GetValueLabel` would be used to convert logarithmic value to DB. This can be a separate PR.

Reply:
I hope its not confusing, that the value parameter of `GetValueLabelFunc` could be the Value from struct or the return value of `GetValueFunc`.

About 4., when `GetValueLabelFunc` is nil, then a [hardcoded float to string conversion](https://github.com/pipelined/vst2/blob/7a0468141ef1aa22304341c155f820b6f368d8a5/vst.go#L778) will be used.

Reply:
> I hope its not confusing, that the value parameter of GetValueLabelFunc could be the Value from struct or the return value of GetValueFunc.

Well, it's up to the user, right? :) They'll get what `GetValue`returns.